catalog/core/catalog-core-ftp/src/main/java/ddf/catalog/core/ftp/FtpServerStarter.java

```diff
@@ -134,6 +195,21 @@ private void suspendServer() {
         }
     }
 
+    private void waitForConnections() {
+        FtpStatistics serverStatistics = ((DefaultFtpServer) server).getServerContext()
+                .getFtpStatistics();
+        while (serverStatistics.getCurrentConnectionNumber() > 0) {
+            LOGGER.warn("Waiting for {} connections to close before updating configuration",
+                    serverStatistics.getCurrentConnectionNumber());
+            try {
+                Thread.sleep(5000);
```

Comment:
We should have this bound to a maximum time to sleep. Otherwise, the whole system could get help up waiting for this connection to end.